src/caikit_nlp_client/http_client.py

```diff
@@ -166,10 +166,15 @@ def generate_text(
             timeout=timeout,
             **req_kwargs,  # type: ignore
         )
-
-        response.raise_for_status()  # TODO
-
-        return response.json()["generated_text"]
+        log.debug(f"Response: {response}")
+        if response.status_code == 200:
+            return response.json()["generated_text"]
+        elif 400 <= response.status_code < 500:
+            raise RuntimeError(response.json()["details"])
+        else:
+            raise RuntimeError(
```

Comment:
```suggestion
        if response.status_code == 200:
            return response.json()["generated_text"]
            
        if 400 <= response.status_code < 500:
            raise RuntimeError(response.json()["details"])
        raise RuntimeError(
```

Reply:
I guess in this case the response could be 200 but we could hit invalid json, which should result in a `ValueError`

Reply:
why break the if ... elif .. else  into if... if else ? Not sure I see the value.